File: lib/listening-minutes.ts
/** Whole minutes from accumulated listen duration (matches JS Math.round). */
export function minutesFromMs(ms: number): number {
  return Math.round(ms / 60_000);
}

/** Whole hours from accumulated listen duration (matches JS Math.round). */
export function hoursFromMs(ms: number): number {
  return Math.round(ms / 3_600_000);
}

File: Soundfolio/Data/ListeningMinutes.swift
import Foundation

enum ListeningMinutes {
    /// Whole minutes from accumulated listen duration. Matches JS `Math.round(ms / 60000)`.
    static func minutes(fromMs ms: Int) -> Int {
        Int((Double(ms) / 60_000).rounded(.toNearestOrAwayFromZero))
    }

    /// Whole hours from accumulated listen duration. Matches JS `Math.round(ms / 3600000)`.
    static func hours(fromMs ms: Int) -> Int {
        Int((Double(ms) / 3_600_000).rounded(.toNearestOrAwayFromZero))
    }
}
